zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/publishing/legacycacheapi/LegacyCacheApiPayloadBuilder.java

```diff
@@ -0,0 +1,205 @@
+                                    if (isValid(legacyCacheApiPayload)) {
+                                        payloadsByPath.put(legacyCacheApiPayload.uriToUpdate, legacyCacheApiPayload);
+                                    }
+                                    addPayloadForBulletinLatest(legacyCacheApiPayload.uriToUpdate, collectionId, this.publishDate);
+                                }
+                        );
+                collection.getDescription().getPendingDeletes()
+                        .forEach(pendingDelete -> {
+                            String uriToDelete = pendingDelete.getRoot().getUri();
+                            uriToDelete = getCanonicalPagePath(uriToDelete, collectionId);
+                            LegacyCacheApiPayload legacyCacheApiPayload = new LegacyCacheApiPayload(collectionId, uriToDelete);
+                            if (isValid(legacyCacheApiPayload)) {
+                                payloadsByPath.put(legacyCacheApiPayload.uriToUpdate, legacyCacheApiPayload);
+                            }
+                        });
+            } catch (IOException e) {
+                error().data("collectionId", collectionId)
+                        .logException(e, "failed initialising PublishNotification when reading collection reviewedUris");
+            }
+
+            return new LegacyCacheApiPayloadBuilder(this.payloadsByPath.values());
+        }
+
+        private void addPayloadForBulletinLatest(String uri, String collectionId, Date clearCacheDate) {
+            if (uri != null && uri.contains(BULLETIN_SEGMENT) && !uri.endsWith("/latest")) {
+                String updatedURI = URIUtils.getNSegmentsAfterSegmentInput(uri, BULLETIN_SEGMENT, 1) + "/latest";
+                LegacyCacheApiPayload legacyCacheApiPayloadLatest = new LegacyCacheApiPayload(collectionId, updatedURI, clearCacheDate);
+                payloadsByPath.put(legacyCacheApiPayloadLatest.uriToUpdate, legacyCacheApiPayloadLatest);
+            }
+        }
+
+        private boolean isValid(LegacyCacheApiPayload payload) {
+            if (StringUtils.isEmpty(payload.uriToUpdate)) {
+                info().data("collectionId", payload.collectionId)
+                        .log("invalid payload: URI was empty or null");
+                return false;
+            }
+            return true;
+        }
+
+        public static String getCanonicalPagePath(String incomingUrl, String  collectionId) {
+            try {
+                if (StringUtils.isEmpty(incomingUrl)) {
+                    return incomingUrl;
+                }
+
+                String updatedUri = incomingUrl.trim();
+                if (StringUtils.isEmpty(updatedUri)) {
+                    return "";
+                }
+
+                // Order matters:
+                // 1. If starts with ons return original
```

Comment:
This isn't actually needed, we don't need to check for ONS paths. An ONS path always returns a long cache time, so the Proxy does not need to query the API in these cases.

Reply:
ok, will remove this